Upgrade OpenLayers

shucon opened the pull request:
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

Fixes: https://github.com/phpmyadmin/phpmyadmin/issues/15296

### Description

Please describe your pull request.

Fixes #

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

shucon committed:
Initial Commit, with display and bounds working

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

mauriciofauth commented on js/table/gis_visualization.js:
```diff
@@ -148,16 +148,17 @@ function initGISVisualization () {
+    var script= document.createElement('script');
+    script.type = 'text/javascript';
+
+    script.src = 'js/vendor/openlayers/OpenLayers.js';
```
Is this necessary? This script tag is created in:
https://github.com/phpmyadmin/phpmyadmin/blob/bcf21c4081232df1bb77a076a3ff551ee7d7b40c/libraries/classes/Controllers/Table/GisVisualizationController.php#L111

mauriciofauth commented on libraries/classes/Gis/GisGeometry.php:
```diff
@@ -139,15 +139,20 @@ abstract class GisGeometry
+//        . $scale_data['maxX'] . ', ' . $scale_data['maxY']
+//        . ').transform(new OpenLayers.Projection("EPSG:'
+//        . intval($srid) . '"), map.getProjectionObject()));';
+        return 'var minLoc = [' . $scale_data['minX'].', '. $scale_data['minY']. ' ];'
+        . 'var maxLoc = [' . $scale_data['maxX'].', '. $scale_data['maxY']. ' ];'
+        . 'var ext = ol.extent.boundingExtent([minLoc, maxLoc]);'
+        . 'ext  = ol.proj.transformExtent(ext, ol.proj.get("EPSG:' . intval($srid) . '"), ol.proj.get(\'EPSG:3857\'));'
+        . 'map.getView().fit(ext, map.getSize());';
```
Use `sprintf` here to improve readability.

shucon replied:
I['m not sure why `sprintf` is required here. It's returning a piece of javascript code.

mauriciofauth replied:
```suggestion
        return sprintf(
            'var minLoc = [%s, %s];'
            . 'var maxLoc = [%s, %s];'
            . 'var ext = ol.extent.boundingExtent([minLoc, maxLoc]);'
            . 'ext = ol.proj.transformExtent(ext, ol.proj.get("EPSG:%s"), ol.proj.get(\'EPSG:3857\'));'
            . 'map.getView().fit(ext, map.getSize());',
            $scale_data['minX'],
            $scale_data['minY'],
            $scale_data['maxX'],
            $scale_data['maxY'],
            intval($srid)
        );
```

shucon replied:
Ah, my bad! I thought `sprintf` was the one used for translations.

shucon committed:
Point Geometry working

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Refactored GIS but not working locally (and no errors)

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Fix MultiPolygon

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Fix Polygon render

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Fix Multipolygon

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Add styling for polygons

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Fix scale

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Add point style

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
LineString style

shucon committed:
Multiline string styling

shucon committed:
Fix extra line bug

williamdes commented on libraries/classes/Gis/GisGeometry.php:
```diff
@@ -418,9 +403,9 @@ protected function getPointsArrayForOpenLayers(array $points_arr, $srid)
-        . '.transform(new OpenLayers.Projection("EPSG:'
-        . intval($srid) . '"), map.getProjectionObject())';
+        return '(new ol.geom.Point([' . $point[0] . ',' . $point[1] . '])'
+        . '.transform(ol.proj.get("EPSG:' . intval($srid) . '")'
```
```suggestion
        . '.transform(ol.proj.get("EPSG:' . ((int) $srid) . '")'
```

williamdes commented on libraries/classes/Gis/GisGeometry.php:
```diff
@@ -418,9 +403,9 @@ protected function getPointsArrayForOpenLayers(array $points_arr, $srid)
-        . intval($srid) . '"), map.getProjectionObject())';
+        return '(new ol.geom.Point([' . $point[0] . ',' . $point[1] . '])'
+        . '.transform(ol.proj.get("EPSG:' . intval($srid) . '")'
+        . ', ol.proj.get(\'EPSG:3857\')))';
```
is it normal we force 3857 here ?

shucon replied:
Yes, that's the projection system used by OpenLayers is 3857. Check: https://epsg.io/3857

MauricioFauth committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 98.5% (3307 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/pt_BR/
Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

BouRock committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (3354 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/tr/
Signed-off-by: Burak Yavuz <hitowerdigit@hotmail.com>

williamdes committed:
Translated using Weblate (French)

Currently translated at 100.0% (3354 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/fr/
Signed-off-by: William Desportes <williamdes@wdes.fr>

shucon committed:
Coding standard

shucon committed:
Add stylesheets

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 58.7% (1971 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/nb_NO/
Signed-off-by: Allan Nordhøy <epost@anotheragency.no>

shucon committed:
Add style to GIS editor

Commit:
Translated using Weblate (Interlingua)

Currently translated at 76.2% (2556 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/ia/
Signed-off-by: Giovanni Sora <g.sora@tiscali.it>

shucon committed:
Merge branch 'master' of github.com:phpmyadmin/phpmyadmin into openlayers

shucon committed:
Mergeable commit

shucon committed:
Change mouse position projection format

shucon committed:
Fix phpstan errors

shucon committed:
Add Zoom buttons and attribution

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

shucon committed:
Fix reclicking to display openlayers bug

Signed-off-by: Saksham Gupta <shucon01@gmail.com>

SantosSi committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (3354 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/pt/
Signed-off-by: ssantos <ssantos@web.de>

Commit:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3354 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/sl/
Signed-off-by: Domen <mitenem@outlook.com>

northy committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (3354 of 3354 strings)

[ci skip]

Translation: phpMyAdmin/Development
Translate-URL: https://hosted.weblate.org/projects/phpmyadmin/master/pt_BR/
Signed-off-by: Alexsandro Thomas <alexsandrogthomas@gmail.com>

shucon committed:
Automated error fixes

shucon committed:
Fix tests

shucon committed:
Fix Automated tests

shucon committed:
Revert "Translated using Weblate (Slovenian)"

This reverts commit aa0e41a08573da4a389f872c04ee6e86bc744656.

shucon committed:
Revert "Translated using Weblate (Portuguese)"

This reverts commit fdb5935972f27887458caf0aad40a61caf4de66c.

shucon committed:
Revert "Translated using Weblate (Portuguese (Brazil))"

This reverts commit dcc8e4cfd9eed6eebaadb32fbfcc740b1181f0d2.

shucon committed:
Revert "Translated using Weblate (Interlingua)"

This reverts commit 11f862f8a568a2a4036501f96c5d58656ef34412.

shucon committed:
Revert "Translated using Weblate (Portuguese (Brazil))"

This reverts commit 463d9f16c0e097c3d75a53f849ff11c91b8505ae.

shucon committed:
Revert "Translated using Weblate (Norwegian Bokmål)"

This reverts commit ef7f823f617e120e1a69c71e99459c8ee8b535d8.

shucon committed:
Revert "Translated using Weblate (French)"

This reverts commit 4dbd9b5f4284fa039cc6aac2bee967f4717d949f.

shucon committed:
Revert "Translated using Weblate (Turkish)"

This reverts commit c59a0a00a81da65d18a7a5cbc75dcdff258d8dd5.

shucon commented:
Closing in reference to: https://github.com/phpmyadmin/phpmyadmin/pull/16303
